### PR TITLE
Initial import of git-bisect article

### PR DESCRIPTION
**Pull request by whereisciao**
Short article on using git bisect for discovering bad commits.

### CODE REVIEW

File: _posts/2022-02-15-uncovering-bad-commits.md [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+Identifying a good commit could be difficult. I tend to select a commit in history a few weeks or months at a time. For this particular scenario, a commit a few weeks ago sufficed.
+
+```
+$> git bisect start
+$> git bisect bad c0fdabe  # A bad commit hash
+$> git bisect good 38ccb50 # A good commit hash
+```
+
+After a few iterations, the problematic commit was discovered. I was able to notify the author of the bug, thus the the issue resolve promptly.
```
Comment (Reddshift) [inline _posts/2022-02-15-uncovering-bad-commits.md:20]: This post is great!  I love that it is short and sweet.  I wonder though if an example or two of what the iterations look like might be helpful?

Comment (stockandawe) [reply _posts/2022-02-15-uncovering-bad-commits.md:20]: Agree with @Reddshift that an example would be great. 

@whereisciao , I think the slides you presented had an example. I don't think there was any code in there that was sensitive and could not be shared.

File: _posts/2022-02-15-uncovering-bad-commits.md [hunk 2]
```diff
@@ -0,0 +1,20 @@
+---
+layout: post
+title: Uncovering Bad Commits
```
Comment (stockandawe) [inline _posts/2022-02-15-uncovering-bad-commits.md:3]: Title suggestion: Finding a needle in a haystack - Uncovering bad commits with git bisect

And suggestion for an image to include in the post - https://unsplash.com/s/photos/needle-haystack

Comment (whereisciao) [reply _posts/2022-02-15-uncovering-bad-commits.md:3]: Title was reworded to include "needed in a haystack". Also grabbed an image.
image

File: _posts/2022-02-15-uncovering-bad-commits.md (hunk 1 above)
Comment (whereisciao) [reply _posts/2022-02-15-uncovering-bad-commits.md:20]: I considered it. I'll add additional commands regarding the iterations.

File: _posts/2022-02-15-finding-a-needle-in-a-haystack.md [hunk 3]
```diff
@@ -0,0 +1,42 @@
+---
+layout: post
+title: Finding a needle in a haystack with git bisect
+author_avatar: https://avatars.githubusercontent.com/u/16843?v=4
+author_name: Steven Chau
+author_github_username: whereisciao
+excerpt: Finding a bad commit in an unfamiliar code base is like finding a needle in a haystack. Determined to find the root cause of the issue, I turned to an essential tool in my tool box, git bisect.
```
Comment (whereisciao) [inline _posts/2022-02-15-finding-a-needle-in-a-haystack.md:7]: I defined an `excerpt` as I preferred the image to be below the title tag, but not have the image included on the index page.

**Comment by whereisciao** [comment]
I'll let this sit for any more feedback. I'll merge the PR on Thursday morning.

**Review by stockandawe — APPROVED** [review]
👍  Nice work. Love a blog post like this that helps any type of developer.

**Pull request merged**